src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/ProvidePublicParameterlessSafeHandleConstructor.Fixer.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    public abstract class ProvidePublicParameterlessSafeHandleConstructorFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(ProvidePublicParameterlessSafeHandleConstructorAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(context.Document);
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode enclosingNode = root.FindNode(context.Span);
```

Comment:
Do we need to pass true for `getInnermostNodeForTie`? If so, we should add a test too.

Reply:
I don't know what that flag does, so I don't know if I need to pass it or what a test case would look like.

Reply:
@jkoritzinsky The flag is when two nodes have the same span, which one to get?
I don't think this can happen for the case of this analyzer, but will let @mavasani confirm in case there is any corner case I'm missing.

Reply:
I don't believe that can happen with the nodes we're looking at in this analyzer, so I don't think we need to worry about it.